pkg/arvo/app/radar.hoon

```diff
@@ -0,0 +1,304 @@
+  ^-  (quip move _this)
+  :-  [[ost.bol %connect / [~ /'~radar'] %radar]]~
+  ?~  old
+    %=    this
+    ::
+        scanning.sty  %.y
+    ::
+        configured.sty  ~
+    ==
+  this(sty u.old)
+::
+++  get-point-json
+  |=  her=ship
+  ^-  json
+  ?.  (~(has in configured.sty) her)
+   [%s 'not being tracked']
+  =/  lastseen/(unit @da)
+    (~(get by outstanding.sty) her)
+  =/  log/(list [@da @da])
+    (fall (~(get by log.sty) her) ~)
+  =/  failed/(list {p/@t q/json})
+    ?~  lastseen
+      ~
+    [%'last-seen' (time:enjs:format u.lastseen)]~
+  %-  pairs:enjs:format
+  :_  failed
+  :-  %responded
+  :-  %a
+  %+  turn  log
+  |=  [sent=@da recv=@da]
+  ^-  json
+  %-  pairs:enjs:format
+  :~  [%sent (time:enjs:format sent)]
+      [%recv (time:enjs:format recv)]
+  ==
+++  poke-noun
+  |=  a=*
+  ^-  (quip move _this)
+  ?+  a
+    [~ this]
+  ::
+      %running
+    ~&  scanning.sty
+    [~ this]
+  ::
+      %print
+    ~&  sty
+    [~ this]
+  ::
+      %start
+    ?:  scanning.sty
+      ~&  'already running'
+      [~ this]
+    =.  this  this(scanning.sty &)
+    ::
+    =.  this  (scan ~(tap in configured.sty))
+    abet
+  ::
+      %stop
+    :_  this(scanning.sty |, timers.sty ~)
+    %-  ~(rep by timers.sty)
+    |=  [[wir=wire d=@da] mov=(list move)]
+    ^-  (list move)
+    [[ost.bol %rest wir d] mov]
+  ::
+      %flush
+    [~ this(log.sty ~)]
+  ==
+::
+
+++  ping
+  |=  her=ship
+  ^+  this
+  %=  this
+  ::
+      moves.sty
+    :_  moves.sty
+    :*  ost.bol
+        %poke
+        /ping/(scot %p her)/(scot %da now.bol)
+        [her %hall]  %noun  'fdsjfk'
+    ==
+  ::
+      outstanding.sty
+    (~(put by outstanding.sty) her now.bol)
+  ==
+::
+++  scan
+  |=  ships=(list ship)
+  ^+  this
+  |-
+  ?~  ships
+    this
+  =.  this  (ping i.ships)
+  $(ships t.ships)
+::
+++  coup
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ?>  ?=([%ping @t @t ~] wir)
+  ::
+  =/  who=@p   (slav %p i.t.wir)
+  =/  wen=@da  (slav %da i.t.t.wir)
+  =/  out=(unit @da)  (~(get by outstanding.sty) who)
+  ?~  out
+    [~ this]    :: no longer tracking ship
+  ?>  =(wen u.out)
+  ::
+  =/  old-log=(unit (list [@da @da]))
+    (~(get by log.sty) who)
+  ::
+  =/  new-log=(list [@da @da])
+    ?~  old-log  [[wen now.bol] ~]
+    [[wen now.bol] u.old-log]
+  ::
+  =.  outstanding.sty  (~(del by outstanding.sty) who)
+
+  ::
+  ?.  scanning.sty
+    [~ this]
+  ::
+  =/  timeout=@da  (add now.bol ~m10)
+  =/  wait-wire=wire  /wait/[i.t.wir]/(scot %da timeout)
+  :-  [ost.bol %wait wait-wire timeout]~
+  %=  this
+  ::
+      log.sty
+    (~(put by log.sty) who new-log)
+  ::
+      timers.sty
+    (~(put by timers.sty) wait-wire timeout)
+  ::
+  ==
+::
+:: Routing
+:: /~radar GET -> list of ships being tracked
+:: /~radar/{shipname} GET -> uptime info of ship
+:: /~radar/{shipname} POST -> start tracking ship (requires authorization)
+:: /~radar/{shipname} DELETE -> stop tracking ship (requires authorization)
```

Comment:
Not sure how useful the POST and DELETE interfaces will be. Probably its easier to control it from dojo, so adding some poke-noun cases for this would be nice. Also I imagine the way this will be used is to track whole classes of ships like all galaxies and stars, so being able to add ships at that level of granularity rather than one at a time would be nice.